homeassistant/components/tasmota_irhvac/climate.py

```diff
@@ -0,0 +1,748 @@
+        vol.Required(CONF_TEMP_SENSOR): cv.entity_id,
+        vol.Optional(
+            CONF_STATE_TOPIC, default=DEFAULT_STATE_TOPIC
+        ): mqtt.valid_subscribe_topic,
+        vol.Optional(CONF_MAX_TEMP, default=DEFAULT_MAX_TEMP): vol.Coerce(float),
+        vol.Optional(CONF_MIN_TEMP, default=DEFAULT_MIN_TEMP): vol.Coerce(float),
+        vol.Optional(CONF_TARGET_TEMP, default=DEFAULT_TARGET_TEMP): vol.Coerce(float),
+        vol.Optional(
+            CONF_INITIAL_OPERATION_MODE, default=DEFAULT_INITIAL_OPERATION_MODE
+        ): vol.In(
+            [STATE_HEAT, STATE_COOL, STATE_AUTO, STATE_DRY, STATE_FAN_ONLY, STATE_OFF]
+        ),
+        vol.Optional(CONF_AWAY_TEMP, default=DEFAULT_AWAY_TEMP): vol.Coerce(float),
+        vol.Optional(CONF_PRECISION, default=DEFAULT_PRECISION): vol.In(
+            [PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE]
+        ),
+        vol.Optional(CONF_MODES_LIST, default=DEFAULT_MODES_LIST): vol.All(
+            cv.ensure_list, [vol.In(HVAC_MODES)]
+        ),
+        vol.Optional(CONF_FAN_LIST, default=DEFAULT_FAN_LIST): vol.All(
+            cv.ensure_list,
+            [
+                vol.In(
+                    [
+                        FAN_ON,
+                        FAN_OFF,
+                        FAN_AUTO,
+                        FAN_LOW,
+                        FAN_MEDIUM,
+                        FAN_HIGH,
+                        FAN_MIDDLE,
+                        FAN_FOCUS,
+                        FAN_DIFFUSE,
+                        HVAC_FAN_MIN,
+                        HVAC_FAN_MEDIUM,
+                        HVAC_FAN_MAX,
+                        HVAC_FAN_AUTO,
+                        HVAC_FAN_MAX_HIGH,
+                        HVAC_FAN_AUTO_MAX,
+                    ]
+                )
+            ],
+        ),
+        vol.Optional(CONF_SWING_LIST, default=DEFAULT_SWING_LIST): vol.All(
+            cv.ensure_list,
+            [vol.In([SWING_OFF, SWING_BOTH, SWING_VERTICAL, SWING_HORIZONTAL])],
+        ),
+        vol.Optional(CONF_QUIET, default=DEFAULT_CONF_QUIET): cv.string,
+        vol.Optional(CONF_TURBO, default=DEFAULT_CONF_TURBO): cv.string,
+        vol.Optional(CONF_ECONO, default=DEFAULT_CONF_ECONO): cv.string,
+        vol.Optional(CONF_MODEL, default=DEFAULT_CONF_MODEL): cv.string,
+        vol.Optional(CONF_CELSIUS, default=DEFAULT_CONF_CELSIUS): cv.string,
+        vol.Optional(CONF_LIGHT, default=DEFAULT_CONF_LIGHT): cv.string,
+        vol.Optional(CONF_FILTER, default=DEFAULT_CONF_FILTER): cv.string,
+        vol.Optional(CONF_CLEAN, default=DEFAULT_CONF_CLEAN): cv.string,
+        vol.Optional(CONF_BEEP, default=DEFAULT_CONF_BEEP): cv.string,
+        vol.Optional(CONF_SLEEP, default=DEFAULT_CONF_SLEEP): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the generic thermostat platform."""
+    name = config.get(CONF_NAME)
```

Comment:
Please use `dict.get(key)` only for optional keys without default value and `dict[key]` for optional keys with default value and required keys.

Reply:
Ok, will do!

Reply:
Done!